include/solvers/post_processors.h

```diff
@@ -166,4 +170,57 @@ class SRFPostprocessor : public DataPostprocessorVector<dim>
   double omega_z;
 };
 
+/**
+ * @brief Calculates de viscosity on each quadrature point for a non Newtonian flow.
+ * The equation of the viscosity depends on the used rheological model.
+ */
+template <int dim>
+class NonNewtonianViscosityPostprocessor : public DataPostprocessorScalar<dim>
+{
+public:
+  NonNewtonianViscosityPostprocessor(
+    Parameters::NonNewtonian p_non_newtonian_parameters)
+    : DataPostprocessorScalar<dim>("viscosity", update_gradients)
+    , non_newtonian_parameters(p_non_newtonian_parameters)
+  {}
+  virtual void
+  evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &inputs,
+                        std::vector<Vector<double>> &computed_quantities) const
+  {
+    Tensor<2, dim>     shear_rate;
+    const unsigned int n_quadrature_points = inputs.solution_gradients.size();
+
+    std::shared_ptr<RheologicalModel<dim>> rheological_model;
+    if (non_newtonian_parameters.model ==
+        Parameters::NonNewtonian::Model::carreau)
+      {
+        rheological_model =
+          std::make_shared<Carreau<dim>>(non_newtonian_parameters);
+      }
```

Comment:
I would create the rheological model in the constructor, not in the call to evaluate vector field.

Reply:
I changed it.